frontend/Movie_Pigeon/app/src/main/java/org/example/team_pigeon/movie_pigeon/RegistrationActivity.java

```diff
@@ -0,0 +1,125 @@
+        Button backButton = (Button) register.findViewById(R.id.rpBackButton);
+
+        // checking whether using third party sign up
+        if (thirdParty.equals("TraktTV")) {
+            etUsername.setText(userInfo);
+        }
+
+        backButton.setOnClickListener(new View.OnClickListener() {
+            @Override
+            public void onClick(View v) {
+                finish();
+            }
+        });
+
+        registerButton.setOnClickListener(new View.OnClickListener() {
+
+            @Override
+            public void onClick(View v) {
+                email = String.valueOf(etEmail.getText());
+                username = String.valueOf(etUsername.getText());
+                password = String.valueOf(etPassword.getText());
+                confirmPassword = String.valueOf(etConfirmPassword.getText());
+
+                if (email.equals("") | username.equals("") | password.equals("") | confirmPassword.equals("")) {
+                    Toast.makeText(getApplicationContext(), "Must fill in all fields", Toast.LENGTH_SHORT).show();
+                } else {
+                    // double check password match
+                    if (!password.equals(confirmPassword)) {
+                        Toast toast = Toast.makeText(getApplicationContext(), "Passwords entered don't match!", Toast.LENGTH_SHORT);
+                        toast.show();
+                    } else {
+                        count = emailChecker(email);
+                        // check whether email field contains: no @, or 2 or more @ or does not have enough length as format a@b
+                        if (count == 0 || count > 1 || email.length()<=2) {
+                            Toast.makeText(getApplicationContext(), "Please enter correct email address", Toast.LENGTH_SHORT).show();
+                        } else {
+                            System.out.println("Registering");
+                            String[] registrationDetails = new String[5];
+                            registrationDetails[0] = email;
+                            registrationDetails[1] = username;
+                            registrationDetails[2] = password;
+                            registrationDetails[3] = thirdParty;
+                            registrationDetails[4] = userInfo;
+
+                            Log.i("RegistrationPage", "5 parameters to be passed are " + registrationDetails[0] + " " + registrationDetails[1] + " " + registrationDetails[2] + " " + registrationDetails[3] + " " + registrationDetails[4]);
+                            new RegistrationHttpBuilder(getApplicationContext()).execute(registrationDetails);
+                        }
+                    }
+                }
+
+            }
+        });
+
+    }
+
+    @Override
+    protected void onDestroy() {
+        super.onDestroy();
+        unregisterReceiver(globalReceiver);
+    }
+
+    private int emailChecker(String email) {
```

Comment:
checker can improve:
1.Only one '@' is in the string.
2.pattern of *.* after @.